### ISSUE TITLE
bgImage css error

### ISSUE THREAD

**Issue by seedev** [fnagel/jquery-ui#146]
on Line 332 is a css-syntax error:

This dont work:
$('#myDiv').css('background', 'path/to/image.jpg');

This will:
$('#myDiv').css('background', 'url(path/to/image.jpg)');

Solution:
thisLi.find('span').css({'background-image': 'url('+selectOptionData[i].bgImage+')'});

**Comment by fnagel** [comment]
Sorry but I did not understand what youre trying to do here. What is #myDiv? Did you take a look in the background_image.html demo file?

**Comment by fnagel** [comment]
Please give some feedback!

**Comment by seedev** [comment]
Ok...

FILE: jquery.ui.selectmenu.js

LINE: 322 ff

``` javascript
[...]
if (o.icons) {
                for (var j in o.icons) {
                    if (thisLi.is(o.icons[j].find)) {
                        thisLi
                            .data('optionClasses', selectOptionData[i].classes + ' ' + self.widgetBaseClass + '-hasIcon')
                            .addClass(self.widgetBaseClass + '-hasIcon');
                        var iconClass = o.icons[j].icon || "";
                        thisLi
                            .find('a:eq(0)')
                            .prepend('<span class="' + self.widgetBaseClass + '-item-icon ui-icon ' + iconClass + '"></span>');
                        if (selectOptionData[i].bgImage) {
                            thisLi.find('span').css('background-image', selectOptionData[i].bgImage);
                        }
                    }
                }
            }
[...]
```

This doesnt work with the   ...  bgImage: callbackfunction ....   Parameter in your constructor !!! Your css-value is wrong

This is yours:

```
thisLi.find('span').css('background-image', selectOptionData[i].bgImage);
```

This is my solution:

```
Try this one: thisLi.find('span').css({'background-image': 'url('+selectOptionData[i].bgImage+')'});
```

```
.css('background-image', selectOptionData[i].bgImage)   --> wrong : 
css({'background-image': 'url('+selectOptionData[i].bgImage+')'})  -> works for me
```

understand???

**Comment by fnagel** [comment]
Please try type 'm' to see some markup infos :-)

Perhaps I should be more precise: please describe what you are trying to archive. Im not sure you understand the mechanism behind the bgImage correct. 

I assume you think of a bug, but its not. We need to make different use cases possible. If you like to use direct parameters try something like this:

```
bgImage: function() {
    return 'url(' + varToYourImg + ')';
}
```

Did you take a look in the background_image.html demo file?

**Comment by fnagel** [comment]
Please give some feedback!

**Comment by seedev** [comment]
Ok.. i see, there are two ways to solve the problem. 
My function returns only a img-src like: 

bgImage: function() {
    return 'picture.jpg';

  // I dont return the url()-syntax around the picture in this function like you. 
 }

Instead of this I modified the other synatx, see comment of August 19th.
And it works, too. Ok - all roads lead to Rome.

 // maybee my solution is better to read, beacuse i dont split the css syntax ?

**Comment by seedev** [comment]
> > Please try type 'm' to see some markup infos :-)

Where?

**Comment by fnagel** [comment]
Mhh this could be a improvement but we're eventually breaking some legacy projects with this change. I plan to change the whole templating mechanism (see: #106). 
Additionally Im currently developing a new version of Selectmenu from scratch which will be ingetrated to the official jQuery UI widgets (see: #140). Currently Im focused on the new version and to fix urgent bugs like this one: #147

So I must confess Im not willing to accept your changes at the moment. Im sorry and be sure your contribution is highly appreciated.
